providers/oidc.go

```diff
@@ -115,11 +118,35 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 	}
 
 	if claims.Email == "" {
-		// TODO: Try getting email from /userinfo before falling back to Subject
-		claims.Email = claims.Subject
+		if p.ProfileURL.String() == "" {
+			return nil, fmt.Errorf("id_token did not contain an email")
+		}
+
+		// If the userinfo endpoint profileURL is defined, then there is a chance the userinfo
+		// contents at the profileURL contains the email.
+		// Make a query to the userinfo endpoint, and attempt to locate the email from there.
+
+		req, err := http.NewRequest("GET", p.ProfileURL.String(), nil)
+		if err != nil {
+			return nil, err
+		}
+		req.Header = getOIDCHeader(token.AccessToken)
+
+		json, err := api.Request(req)
```

Comment:
I'm troubled by the `json` name here. Since this is the JSON from the response body, how do you feel about renaming this to `respJSON`?

My main problem is that it is common to import the package `json` as `json`... and this is a really easy clash we could make later (so easy to shadow and not notice), WDYT?

Reply:
Good call.  I'll make this change.